app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -31,54 +38,51 @@ protected void onCreate(Bundle savedInstanceState) {
         binding = ActivityFragmentJavaBinding.inflate(getLayoutInflater());
         setContentView(binding.getRoot());
 
-        if(savedInstanceState == null) {
-            WebUiFragment fragment = new WebUiFragment();
-            Bundle bundle = new Bundle();
-
-            // Pass in a navigation override. Defaults as .DASHBOARD
-            bundle.putParcelable(
-                    WebUiFragment.ARG_NAVIGATION_OVERRIDE,
-                    NavigationIntent.DASHBOARD.INSTANCE
-            );
-
-            // Pass in a custom Theme override. Defaults to null
-            WebTheme theme = new WebTheme(new Brand());
-            bundle.putParcelable(WebUiFragment.ARG_WEB_THEME_OVERRIDE, theme);
-
-            fragment.setArguments(bundle);
-
-            getSupportFragmentManager()
-                    .beginTransaction()
-                    .replace(binding.fragmentContainer.getId(), fragment, "FRAGMENT_TAG")
-                    .commit();
+        if (savedInstanceState == null) {
+            loadWebUi();
         }
     }
 
     /**
      * Required Override when using [WebUiFragment.Listener]
-     *
+     * <p>
      * Handle when the network error fragment needs to be navigated to. We have provided a
      * [WebUiNetworkErrorFragment] that you can display to the user. Or you can use your
      * own fragment or even an alternative logic flow if desired.
      */
     @Override
     public void onDisplayNetworkError(boolean networkNotAvailable) {
-        WebUiNetworkErrorFragment fragment = new WebUiNetworkErrorFragment();
-        Bundle bundle = new Bundle();
+        // If the fragment manager already has reference to an error fragment, reuse that instance.
+        // Otherwise, create a fresh one.
+        Fragment existingFragment = findFragmentByTag(TAG_WEB_NETWORK_ERROR_FRAGMENT);
+        WebUiNetworkErrorFragment fragment = existingFragment instanceof WebUiNetworkErrorFragment ?
+                (WebUiNetworkErrorFragment) existingFragment :
+                new WebUiNetworkErrorFragment();
+
+        Bundle args = new Bundle();
 
         // If not provided, the fragment assumes that there is no network connection (default: true)
-        bundle.putBoolean(WebUiNetworkErrorFragment.ARG_NO_NETWORK_ERROR, networkNotAvailable);
-        fragment.setArguments(bundle);
+        args.putBoolean(WebUiNetworkErrorFragment.ARG_NO_NETWORK_ERROR, networkNotAvailable);
+        fragment.setArguments(args);
 
-        getSupportFragmentManager()
-                .beginTransaction()
-                .replace(binding.fragmentContainer.getId(), fragment, "NETWORK_ERROR_FRAGMENT_TAG")
-                .commit();
+        showFragmentInContainer(fragment, TAG_WEB_NETWORK_ERROR_FRAGMENT);
+    }
+
+    @Override
+    public void onBackPressed() {
+        Fragment existingFragment = findFragmentByTag(TAG_WEB_NETWORK_ERROR_FRAGMENT);
+        if (existingFragment instanceof WebUiFragment) {
```

Comment:
can we combine these ifs so that we don't have to write `super.onBackPressed()` in two separate conditions?

Reply:
adjusted